### PR TITLE
throw error when rng sampler is fed empty array

### LINKED ISSUE #46272: Bad error message in rand([])
```julia
julia> rand([])
ERROR: ArgumentError: range must be non-empty
Stacktrace:
 [1] SamplerRangeNDL
   @ ~/.julia/juliaup/julia-1.8.0-rc3+0.x64/share/julia/stdlib/v1.8/Random/src/generation.jl:333 [inlined]
 [2] Sampler
   @ ~/.julia/juliaup/julia-1.8.0-rc3+0.x64/share/julia/stdlib/v1.8/Random/src/generation.jl:189 [inlined]
 [3] Sampler
   @ ~/.julia/juliaup/julia-1.8.0-rc3+0.x64/share/julia/stdlib/v1.8/Random/src/generation.jl:418 [inlined]
 [4] Sampler
   @ ~/.julia/juliaup/julia-1.8.0-rc3+0.x64/share/julia/stdlib/v1.8/Random/src/Random.jl:140 [inlined]
 [5] rand
   @ ~/.julia/juliaup/julia-1.8.0-rc3+0.x64/share/julia/stdlib/v1.8/Random/src/Random.jl:254 [inlined]
 [6] rand(X::Vector{Any})
   @ Random ~/.julia/juliaup/julia-1.8.0-rc3+0.x64/share/julia/stdlib/v1.8/Random/src/Random.jl:259
 [7] top-level scope
   @ REPL[273]:1
```
`[]` is not a range.

### PR DESCRIPTION
**Pull request by Moelf**
fix #46272

### CODE REVIEW

**Comment by rikhuijzer** [comment]
Shouldn't there be a test?

**Comment by rfourquet** [comment]
Note that `rand(())` would still throw the same "range must be non-empty" error. What about changing the existing wording to "collection must be non-empty", which will still be correct for ranges, and will be identical to the message for empty dicts and sets. (Also it will avoid to make this check twice, although the cost is probably negligible).

**Comment by Moelf** [comment]
The sampler API which dispatch on Array should just say Array right?

**Comment by ViralBShah** [comment]
@rfourquet Please merge whenever you think ok.

File: stdlib/Random/src/generation.jl [hunk 1]
```diff
@@ -210,7 +210,7 @@ SamplerRangeFast(r::AbstractUnitRange{T}) where T<:BitInteger =
     SamplerRangeFast(r, uint_sup(T))
 
 function SamplerRangeFast(r::AbstractUnitRange{T}, ::Type{U}) where {T,U}
-    isempty(r) && throw(ArgumentError("range must be non-empty"))
+    isempty(r) && throw(ArgumentError("collection must be non-empty"))
```
Comment (rfourquet) [inline stdlib/Random/src/generation.jl:213]: This fixes the problem when the array has indexes of type `Int128`, but not the main case of `Array`. Other instances of `"range must be non-empty"` should also be updated.

**Review by rfourquet — APPROVED** [review]
LGTM, can you check the CI errors are all unrelated?

**Pull request merged**